Rules/AvoidUnInitializedVarsInNewRunspaces.cs

```diff
@@ -0,0 +1,363 @@
+                    ruleId: variableExpression.ToString());
+            }
+        }
+
+        private static IEnumerable<VariableExpressionAst> ProcessInvokeCommandSessionAsts(ScriptBlockAst scriptBlockAst)
+        {
+            var invokeCommandCmdletNamesAndAliases = Helper.Instance.CmdletNameAndAliases("Invoke-Command");
+            var sessionDictionary = new Dictionary<string, List<Ast>>();
+            var result = new List<VariableExpressionAst>();
+
+            // The shortest unambiguous name for parameter -Session is 'session' (SessionName and SessionOption) also exist.
+            var scriptBlockExpressionAstsToAnalyze = scriptBlockAst.FindAll(
+                predicate: a => a is ScriptBlockExpressionAst scriptBlockExpressionAst &&
+                                scriptBlockExpressionAst.Parent is CommandAst commandAst &&
+                                invokeCommandCmdletNamesAndAliases
+                                    .Contains(commandAst.GetCommandName(), StringComparer.OrdinalIgnoreCase) &&
+                                commandAst.CommandElements.Any(
+                                    e => e is CommandParameterAst parameterAst &&
+                                         parameterAst.ParameterName.Equals(
+                                             "session",
+                                             StringComparison
+                                                 .OrdinalIgnoreCase)), 
+                searchNestedScriptBlocks: false);
+
+            // Match ScriptBlocks together that belong to the same session
+            foreach (var ast in scriptBlockExpressionAstsToAnalyze)
+            {
+                if (!(ast.Parent is CommandAst commandAst))
+                    continue;
+
+                // Find session parameter
+                //   The shortest unambiguous name for parameter -Session is 'session' (SessionName and SessionOption) also exist.
+                if (!(commandAst.CommandElements.First<Ast>(
+                        e => e is CommandParameterAst parameterAst &&
+                             parameterAst.
+                                 ParameterName.
+                                 Equals(
+                                     "session",
+                                     StringComparison
+                                         .OrdinalIgnoreCase)) is CommandParameterAst sessionParameterAst))
+                    continue;
+
+                // Extract session name from session parameter
+                string sessionName;
+                try
+                {
+                    sessionName = commandAst
+                        .CommandElements[
+                            commandAst
+                                .CommandElements
+                                .IndexOf(sessionParameterAst) + 1]
+                        .Extent
+                        .Text
+                        .Trim();
+                }
+                catch

```

Comment:
My reasoning was, that if -Session is used without an argument (and the runtime would allow it), it would be OK to just catch and continue. It a broken situation anyways. 
What would be a better way to convey this? Maybe like this:
```csharp
if (commandAst.CommandElements.Count > commandAst.CommandElements.IndexOf(sessionParameterAst)) {
    sessionName = commandAst
                        .CommandElements[
                            commandAst
                                .CommandElements
                                .IndexOf(sessionParameterAst) + 1]
                        .Extent
                        .Text
                        .Trim();
}
```

Reply:
I've refactored it a bit.